pkg/istio-agent/xds_proxy.go

```diff
@@ -386,7 +386,7 @@ func (p *XdsProxy) HandleUpstream(ctx context.Context, con *ProxyConnection, xds
 				proxyLog.Warnf("upstream terminated with unexpected error %v", err)
 				metrics.IstiodConnectionErrors.Increment()
 			}
-			return nil
+			return err
```

Comment:
I think we want to only return unexpected error

Reply:
I think Envoy does not distinguish between unexpected vs expected errors. it just retries https://github.com/envoyproxy/envoy/blob/main/source/common/config/grpc_stream.h#L100

Without this change, we will loose logging in Envoy

Reply:
lg